pinhole.hpp compiler error with std::max

<!--
First look if there is already a similar bug report. If there is, upvote the issue with 👍
-->

**Describe the bug**
Compiler error pointing to std::max function in [pinhole.hpp](https://github.com/rerun-io/rerun/blob/0.14.1/rerun_cpp/src/rerun/archetypes/pinhole.hpp#L156)
```
const float focal_length_y = 0.5f / std::tan(std::max(fov_y * 0.5f, EPSILON));
```
Compiler errors:
```
[build] XXX\rerun\archetypes/pinhole.hpp(156,63): error C2589: '(': illegal token on right side of '::' (compiling source file YYY.cpp) [ZZZ.vcxproj]
[build] XXX\rerun\archetypes/pinhole.hpp(156): error C2144: syntax error: 'unknown-type' should be preceded by ')' (compiling source file YYY.cpp) [ZZZ.vcxproj]
[build] XXX\rerun\archetypes/pinhole.hpp(156,1): error C2661: 'tan': no overloaded function takes 0 arguments (compiling source file YYY.cpp) [ZZZ.vcxproj]
[build] XXX\rerun\archetypes/pinhole.hpp(156,25): error C2737: 'focal_length_y': const object must be initialized (compiling source file YYY.cpp) [ZZZ.vcxproj]
[build] XXX\rerun\archetypes/pinhole.hpp(156): error C2144: syntax error: 'unknown-type' should be preceded by ';' (compiling source file YYY.cpp) [ZZZ.vcxproj]
[build] XXX\rerun\archetypes/pinhole.hpp(156,63): error C2143: syntax error: missing ')' before '*' (compiling source file YYY.cpp) [ZZZ.vcxproj]
[build] XXX\rerun\archetypes/pinhole.hpp(156,63): error C2143: syntax error: missing ';' before '*' (compiling source file YYY.cpp) [ZZZ.vcxproj]
[build] XXX\rerun\archetypes/pinhole.hpp(156,1): error C2082: redefinition of formal parameter 'fov_y' (compiling source file YYY.cpp) [ZZZ.vcxproj]
[build] XXX\rerun\archetypes/pinhole.hpp(156,1): error C2059: syntax error: ')' (compiling source file YYY.cpp) [ZZZ.vcxproj]
[build] XXX\rerun\archetypes/pinhole.hpp(156,63): error C2100: illegal indirection (compiling source file YYY.cpp) [ZZZ.vcxproj]
[build] Build finished with exit code 1
```

**To Reproduce**
Steps to reproduce the behavior:
1. Compile C++ project linking Rerun

**Expected behavior**
No compiler errors

**Desktop (please complete the following information):**
 - OS: Windows, VS 2019

**Rerun version**
0.14.1

**Additional context**
Workaround is manually patch this line by adding braces:
```
const float focal_length_y = 0.5f / std::tan((std::max)(fov_y * 0.5f, EPSILON));
```

--- a/rerun_cpp/src/rerun/archetypes/pinhole.hpp
+++ b/rerun_cpp/src/rerun/archetypes/pinhole.hpp
@@ -153,7 +153,8 @@ namespace rerun::archetypes {
         /// Assumes the principal point to be in the middle of the sensor.
         static Pinhole from_fov_and_aspect_ratio(float fov_y, float aspect_ratio) {
             const float EPSILON = std::numeric_limits<float>::epsilon();
-            const float focal_length_y = 0.5f / std::tan(std::max(fov_y * 0.5f, EPSILON));
+            // `max` has explicit template args to avoid preprocessor replacement when <windows.h> is included without NOMINMAX.
+            const float focal_length_y = 0.5f / std::tan(std::max<float>(fov_y * 0.5f, EPSILON));
             return from_focal_length_and_resolution(
                 {focal_length_y, focal_length_y},
                 {aspect_ratio, 1.0}

--- a/rerun_cpp/src/rerun/archetypes/pinhole_ext.cpp
+++ b/rerun_cpp/src/rerun/archetypes/pinhole_ext.cpp
@@ -36,7 +36,8 @@ namespace rerun {
         /// Assumes the principal point to be in the middle of the sensor.
         static Pinhole from_fov_and_aspect_ratio(float fov_y, float aspect_ratio) {
             const float EPSILON = std::numeric_limits<float>::epsilon();
-            const float focal_length_y = 0.5f / std::tan(std::max(fov_y * 0.5f, EPSILON));
+            // `max` has explicit template args to avoid preprocessor replacement when <windows.h> is included without NOMINMAX.
+            const float focal_length_y = 0.5f / std::tan(std::max<float>(fov_y * 0.5f, EPSILON));
             return from_focal_length_and_resolution(
                 {focal_length_y, focal_length_y},
                 {aspect_ratio, 1.0}

--- a/rerun_cpp/tests/windows_min_max_robustness.cpp
+++ b/rerun_cpp/tests/windows_min_max_robustness.cpp
@@ -1,0 +1,10 @@
+
+// `<windows.h>` by default defined min and max as macros, which can cause issues with std::min and std::max.
+// This can be easily turned off by the user with `#define NOMINMAX` before including `<windows.h>`.
+// However, users might not know about this or forget about it and we want to be robust against it.
+//
+// This test checks that rerun.h is robust against this issue by explicitly setting `min`/`max` macros and including rerun.hpp.
+#define min(a, b) (this does not compile)
+#define max(a, b) (this does not compile)
+
+#include <rerun.hpp>

